app/task_routes.py

```diff
@@ -0,0 +1,134 @@
+            ))
+
+    task = Task.from_dict(request_body)
+
+    db.session.add(task)
+    db.session.commit()
+
+    return jsonify({"task": task.to_dict()}), 201
+
+###### Read Route ######
+
+@tasks_bp.route("", methods=["GET"])
+def read_tasks():
+    
+    title_query = request.args.get("sort")
+    if title_query == "asc":
+        tasks = Task.query.order_by(Task.title).all()
+    elif title_query == "desc":
+        tasks = Task.query.order_by(Task.title.desc()).all()
+    else:
+        tasks = Task.query.all()
+
+    tasks_response = []
+    for task in tasks:
+        tasks_response.append(task.to_dict()), 200
+
+    return jsonify(tasks_response)
+
+@tasks_bp.route("/<task_id>", methods=["GET"])
+def read_one_saved_task(task_id):
+    task = validate_model(Task, task_id)
+
+    return {"task": task.to_dict()}
+    
+###### Update Route ######
+@tasks_bp.route("/<task_id>", methods=["PUT"])
+def update_task(task_id):
+    task = validate_model(Task, task_id)
+
+    request_body = request.get_json()
+
+    task.title = request_body["title"]
+    task.description = request_body["description"]
+
+    db.session.commit()
+
+    return make_response({"task": task.to_dict()},200)
+
+###### Patch Route ######
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def mark_task_complete(task_id):
+    task = validate_model(Task, task_id)
+
+    request_body = request.get_json()
+
+    task.completed_at = datetime.datetime.now()
+
+    db.session.commit()
+
+    slack_channel_id = "C0581Q2TDGQ"
+    slack_url = "https://slack.com/api/chat.postMessage"
+
+    slack_data = {
+        "channel": slack_channel_id,
+        "text": f"Someone completed the {task.title} task!"
+    }
+
+    headers= {
+        "Authorization": f"Bearer {os.environ.get('SLACK_KEY')}"
+    }
+    
+    slack_request = requests.post(slack_url, headers=headers, json=slack_data)
+    print(slack_request.text)
+    return make_response({"task": task.to_dict()}), 200
+
+@tasks_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def mark_task_incomplete(task_id):
+    task = validate_model(Task, task_id)
+
+    request_body = request.get_json()
+
+    task.completed_at = None
+
+    db.session.commit()
+                
+    return make_response({"task": task.to_dict()}, 200)
+
+###### Delete Route ######
+
+@tasks_bp.route("/<task_id>", methods=["DELETE"])
+def delete_task(task_id):
+    task = validate_model(Task, task_id)
+
+    db.session.delete(task)
+    db.session.commit()
+
+    return make_response(
+        {
+            "details": f"Task {task.task_id} \"{task.title}\" successfully deleted"
+        }, 200
+    )
+
+def validate_model(cls, model_id):
+    try:
+        model_id = int(model_id)
+    except:
+        abort(make_response({"details":f"Invalid data"}, 400))
+
+    model = cls.query.get(model_id)
+
+    if not model:
+        abort(make_response({"details":f"{cls.__name__} ID not found"}, 404))
+
+    return model
```

Comment:
Make sure that your helper functions are in a place that is easy to find, I typically put them in the top of the file or in another folder somewhere else.

Reply:
Well done on this project, I didn't have much to comment on and that is a good thing! Keep up the good work! Really looking forward to what you create in the frontend! Please feel free to reach out if you have any questions about the feedback that I left! ✨💫🤭